test/aio5gc/msg/nas/handler/authenticationResponse.go

```diff
@@ -9,15 +9,38 @@ import (
 	"errors"
 	"my5G-RANTester/test/aio5gc/context"
 	"my5G-RANTester/test/aio5gc/msg"
+	"my5G-RANTester/test/aio5gc/state"
 	"strings"
 
+	"fmt"
+
 	"github.com/free5gc/nas"
 	"github.com/free5gc/ngap/ngapType"
-
+	"github.com/free5gc/util/fsm"
 	log "github.com/sirupsen/logrus"
 )
 
 func AuthenticationResponse(nasMsg *nas.Message, gnb *context.GNBContext, ue *context.UEContext, amf *context.AMFContext) error {
+	var err error
+	switch ue.GetState().Current() {
+	case state.AuthenticationInitiated:
+		err = DefaultAuthenticationResponse(nasMsg, gnb, ue, amf)
+	case state.Deregistrated:
```

Comment:
default:
return fmt.Errorf("[5GC][NAS] Unexpected message: received %s for Authenticated UE", ue.GetState().Current())
With something to convert your state as a string if required, no need for this big switch for the default cases in each of the states